lib/msf/core/modules/metadata/cache.rb

```diff
@@ -91,12 +106,27 @@ def get_unchanged_module_references
   private
   #######
 
+  def remove_from_cache(module_name)
+    removed = @module_metadata_cache.delete_if {|_, module_metadata|
```

Comment:
The `delete_if` method doesn't appear to return a hash of deleted key-value pairs so I'm uncertain this will function as you intend.

```
$ irb
irb(main):001:0> h = { "a" => 100, "b" => 200, "c" => 300 }
=> {"a"=>100, "b"=>200, "c"=>300}
irb(main):002:0> h.delete_if {|key, value| key >= "b" }
=> {"a"=>100}
irb(main):003:0> h
=> {"a"=>100}
```

Reply:
good catch